wrangler-service/src/main/java/co/cask/wrangler/service/gcs/GCSService.java

```diff
@@ -387,8 +388,10 @@ public void loadObject(HttpServiceRequest request, HttpServiceResponder responde
           }
 
           List<Row> rows = new ArrayList<>();
-          for (String line : lines) {
-            rows.add(new Row("body", line));
+          // if the content was truncated, ignore the last line because it's probably not complete.
```

Comment:
do we need to worry if the file actually has large number of columns, that the partial data in the blob is actually still within the same row?

Reply:
this only truncates if there is more than 1 row. I suppose this may look funny if there were one and a half rows or something, though I think we'll have other issues if somebody has a 5mb row.